default/python/AI/InvasionAI.py

```diff
@@ -76,15 +78,25 @@ def get_invasion_fleets():
             planet_partial_vis_turn = universe.getVisibilityTurnsMap(pid, empire_id).get(fo.visibility.partial, -9999)
             if planet_partial_vis_turn < sys_partial_vis_turn:
                 continue
+            best_base_planet = INVALID_ID
+            best_trooper_count = 1  # won't bother using base troopers unless better than 1 per ship
             for pid2 in state.get_empire_inhabited_planets_by_system().get(sys_id, []):
                 if available_pp.get(pid2, 0) < 2:  # TODO: improve troop base PP sufficiency determination
                     break
                 planet2 = universe.getPlanet(pid2)
                 if not planet2:
                     continue
-                if pid not in foAI.foAIstate.qualifyingTroopBaseTargets and planet2.speciesName in ColonisationAI.empire_ship_builders:
-                    foAI.foAIstate.qualifyingTroopBaseTargets.setdefault(pid, [pid2, INVALID_ID])
-                    break
+                if planet2.speciesName in ColonisationAI.empire_ship_builders:
+                    best_base_trooper_here = \
+                        ProductionAI.get_best_ship_info(PriorityType.PRODUCTION_ORBITAL_INVASION, pid2)[1]
+                    troops_per_ship = best_base_trooper_here.troopCapacity
+                    species_troop_grade = CombatRatingsAI.get_species_troops_grade(planet2.speciesName)
+                    troops_per_ship = CombatRatingsAI.weight_attack_troops(troops_per_ship, species_troop_grade)
+                    if troops_per_ship > best_trooper_count:
+                        best_base_planet = pid2
+                        best_trooper_count = troops_per_ship
+            if best_base_planet != INVALID_ID:
+                foAI.foAIstate.qualifyingTroopBaseTargets.setdefault(pid, [best_trooper_count, INVALID_ID])
```

Comment:
You're right that it is/was obscure, and unnecessarily complex.  I've added comments explaining it and proposing potential simplifications.  My changing thoughts over time as to which values to use and how, account for the discrepancy you note, but I still don't really make use of the two values the way I envision eventually doing, and until such time it should perhaps just drop to one value.  But rather than do that right away I wanted to make sure I got a related bugfix in right away for anyone testing this and mull over the rest a bit more.